backend/api/routes/connections.js

```diff
@@ -225,10 +139,83 @@ router.route("/send-request").post(authenticateMiddleware, async (req, res) => {
+      .from("user_weights")
+      .select("*")
+      .eq("user_id", sender_id)
+      .single();
+
+    //remember calcAgeScore returns 2 or 4 if ages are closer so if its greater than 1 we consider it that user cares about age
+    const ageClose = calcAgeScore(user, streamer) > WEIGHT_UPDATE_TRESHOLD;
+    //same thing language matches if calc language checks them and they return greater than 1 cause they are same
+    const languageMatch =
+      calcLanguageScore(user, streamer) > WEIGHT_UPDATE_TRESHOLD;
+    const gameMatch = calcGameScore(user, streamer) > WEIGHT_UPDATE_TRESHOLD;
+
+    //now after we checked why user connected with this streamer. Did he give importance to languages matching, games matching, age?
+    //we update all the weights based on the matching
+    if (ageClose) userWeights.age_weight += WEIGHT_INCREASE;
+    if (gameMatch) userWeights.game_weight += WEIGHT_INCREASE;
+    if (languageMatch) userWeights.language_weight += WEIGHT_INCREASE;
+
+    //Below two weight updates since they arent something as binary as same language or range of age we update based on user preference
+    //dont need to parse audience_preferences I forgot supabase returns it as an object already not a string
+    const audiencePreferences = userWeights.audience_preferences;
+    const streamerAudience = streamer.targetAudience; //get what audience the streamer streams too
+
+    //since when users sign up they have a dropdown list of audiences streamers audience will always exist in our JSON of audience preference weights
+    //increase the weight of that streamer's audience in our preferences because if we connected with a mature streamer it is because we want to see more of those.
+    audiencePreferences[streamerAudience] += WEIGHT_INCREASE;
+
+    const noDupTags = mergeAndDeduplicateTags(
+      streamer.tags,
+      streamer.twitch_tags
+    );
+
+    //for each tag the streamer we connected with has
+    noDupTags.forEach((tag) => {
+      if (userWeights.preferred_tags[tag]) {
+        //check the json of preffered tags if that tag exists in there already update the weight on that tag
+        userWeights.preferred_tags[tag] += WEIGHT_INCREASE;
+      } else {
+        //if the tag doesnt exist then add it as a new tag and add the intial boost of 1.1 that is what all new tags will start with
+        userWeights.preferred_tags[tag] = 1.1;
+      }
+    });
+
+    //now that everything is updated accordingly actually save these weights in our database
+    await req.supabase
+      .from("user_weights")
+      .update({
+        age_weight: userWeights.age_weight,
+        audience_preferences: audiencePreferences,
+        game_weight: userWeights.game_weight,
+        language_weight: userWeights.language_weight,
+        preferred_tags: userWeights.preferred_tags,
+      })
+      .eq("user_id", sender_id);
+
+    res.status(201).json({
+      success: true,
+      message: "Connection request sent and weigths updated",
+    });
```

Comment:
Just to be crystal clear, the idea is not block the initial call, making the update weights BEFORE returning will only add latency to your call when there's no reason to do that. The [example](https://stackoverflow.com/questions/70502851/how-can-i-execute-a-promise-after-the-endpoint-response/70502880#70502880) provided shows you how to schedule a promise (i.e. not blocking the main call), but here you're blocking the main call making synchronous calls.